SWLOR.Game.Server/Feature/TrapDefinition/PitfallTrap.cs

```diff
@@ -0,0 +1,36 @@
+﻿using SWLOR.Game.Server.Core;
+using SWLOR.Game.Server.Core.NWNX;
+using SWLOR.Game.Server.Core.NWScript.Enum.Creature;
+
+namespace SWLOR.Game.Server.Feature
+{
+    public class PitfallTrap
+    {
+        /// <summary>
+        /// When this trap is triggered, it'll toss the players to the designated waypoint after a delay.
+        /// </summary>
+        [NWNEventHandler("OnTrapTriggered")]
+
+        public static void TrapTriggered()
+
+        {
+            if (GetTag(OBJECT_SELF) != "PitfallTrap")
```

Comment:
There is no reason to check the tag name. This script should be specific to the trigger you've created. Set the OnEnter script of the trigger to the name of this script (after you fix it above) and remove this check.